cpp/oneapi/dal/backend/primitives/objective_function/test/logloss_dpc.cpp

```diff
@@ -147,57 +166,75 @@ class logloss_test : public te::float_algo_fixture<Param> {
         auto [out_logloss, out_e] =
             ndarray<float_t, 1>::zeros(this->get_queue(), { 1 }, sycl::usm::alloc::device);
         sycl::event logloss_event = compute_logloss(this->get_queue(),
-                                                    params_gpu,
-                                                    data_gpu,
                                                     labels_gpu,
+                                                    out_predictions,
                                                     out_logloss,
-                                                    L1,
-                                                    L2,
                                                     fit_intercept,
                                                     { out_e });
-        logloss_event.wait_and_throw();
+        sycl::event logloss_reg_event = add_regularization_loss(this->get_queue(),
+                                                                params_gpu,
+                                                                out_logloss,
+                                                                L1,
+                                                                L2,
+                                                                fit_intercept,
+                                                                { logloss_event });
+        logloss_reg_event.wait_and_throw();
```

Comment:
I believe you can pass event as a dependency to the `to_host` function or rather to `at_device` function